processor/otel/consumer.go

```diff
@@ -58,3 +74,437 @@ func (c *Consumer) ConsumeTraceData(ctx context.Context, td consumerdata.TraceDa
+		switch v := v.Value.(type) {
+		case *tracepb.AttributeValue_BoolValue:
+			utility.DeepUpdate(labels, k, v.BoolValue)
+			if k == "error" {
+				hasFailed = v.BoolValue
+			}
+		case *tracepb.AttributeValue_DoubleValue:
+			utility.DeepUpdate(labels, k, v.DoubleValue)
+		case *tracepb.AttributeValue_IntValue:
+			switch kDots {
+			case "http.status_code":
+				intv := int(v.IntValue)
+				http.Response = &model.Resp{StatusCode: &intv}
+				result = statusCodeResult(intv)
+				isHTTP = true
+			default:
+				utility.DeepUpdate(labels, k, v.IntValue)
+			}
+		case *tracepb.AttributeValue_StringValue:
+			switch kDots {
+			case "span.kind": // filter out
+			case "http.method":
+				http.Request = &model.Req{Method: truncate(v.StringValue.Value)}
+				isHTTP = true
+			case "http.url", "http.path":
+				event.Url = parseURL(v.StringValue.Value, hostname)
+				isHTTP = true
+			case "http.status_code":
+				if intv, err := strconv.Atoi(v.StringValue.Value); err == nil {
+					http.Response = &model.Resp{StatusCode: &intv}
+					result = statusCodeResult(intv)
+				}
+				isHTTP = true
+			case "http.protocol":
+				if strings.HasPrefix(v.StringValue.Value, "HTTP/") {
+					version := truncate(strings.TrimPrefix(v.StringValue.Value, "HTTP/"))
+					http.Version = &version
+				} else {
+					utility.DeepUpdate(labels, k, v.StringValue.Value)
+				}
+				isHTTP = true
+			case "type":
+				event.Type = truncate(v.StringValue.Value)
+			case "component":
+				component = truncate(v.StringValue.Value)
+				fallthrough
+			default:
+				utility.DeepUpdate(labels, k, truncate(v.StringValue.Value))
+			}
+		}
+	}
+
+	if event.Type == "" {
+		if isHTTP {
+			event.Type = "request"
+		} else if component != "" {
+			event.Type = component
+		} else {
+			event.Type = "custom"
+		}
+	}
+
+	if isHTTP {
+		if code := int(span.GetStatus().GetCode()); code != 0 {
+			result = statusCodeResult(code)
+			if http.Response == nil {
+				http.Response = &model.Resp{StatusCode: &code}
+			}
+		}
+		event.Http = &http
+	}
+
+	if result == "" {
+		if hasFailed {
+			result = "Error"
+		} else {
+			result = "Success"
+		}
+	}
+	event.Result = &result
+
+	if len(labels) == 0 {
+		return
+	}
+	l := model.Labels(labels)
+	event.Labels = &l
+}
+
+func parseSpan(span *tracepb.Span, event *model_span.Event) {
+	labels := make(common.MapStr)
+
+	var http model_span.HTTP
+	var db model_span.DB
+	var destination model_span.Destination
+	var isDBSpan, isHTTPSpan bool
+	var component string
+	for kDots, v := range span.Attributes.GetAttributeMap() {
+		k := replaceDots(kDots)
+		switch v := v.Value.(type) {
+		case *tracepb.AttributeValue_BoolValue:
+			utility.DeepUpdate(labels, k, v.BoolValue)
+		case *tracepb.AttributeValue_DoubleValue:
+			utility.DeepUpdate(labels, k, v.DoubleValue)
+		case *tracepb.AttributeValue_IntValue:
+			switch kDots {
+			case "http.status_code":
+				//v.IntValue is int64, considering the semantics of the field converting to an int should be fine
```

Comment:
I think it's safe to remove this comment, code seems pretty logical to me and we do the same thing in several other places.